Modules/Module_Processing/Module_ASL/xASL_wrp_Quantify.m

```diff
@@ -109,59 +115,19 @@ function xASL_wrp_Quantify(x, PWI4D_Path, pathOutputCBF, M0Path, SliceGradientPa
 %% 1.   Load PWI4D
 fprintf('%s\n','Loading PWI4D & M0 images');
 
-%% #1543 This goes to xASL_module_ASL
-% Either load a PWI image or merge several PWIs
-if x.modules.asl.bMergingSessions == 1
-	PWI4D = xASL_wrp_Quantify_MergeSessions(x); 
-    %% #1543 THIS WILL BE RENAMED TO xASL_im_MergePWI4D
-    %% AND WILL BE RUN SEPARATELY BEFORE THIS WRAPPER
-    %% TO COMBINE BOTH IMAGES AND JSON, AND SAVE THEM
-    %
-    %% PM: LOAD JSON HERE AS WELL, USING xASL_io_Nifti2Im
-else
-	% Load ASL single PWI
-	[PWI4D, JSON] = xASL_io_Nifti2Im(PWI4D_Path); % Load CBF nifti
-end
-
-if isempty(JSON)
-    %% PM: #1543 for backward compatibility, if part of the ASL module was run
-    %% we load ASL_parms or use the x.Q
-	warning(['Reverting to x.Q memory, JSON file missing for: ' PWI4D_Path]);
-    ASL_parms = xASL_adm_LoadParms(x.P.Path_ASL4D_parms_mat, x);
+% Load ASL single PWI
+[~, jsonASL] = xASL_io_Nifti2Im(PWI4D_Path); % Load CBF nifti
+jsonASL = xASL_bids_parms2BIDS([], jsonASL, 0); % BIDS to Legacy conversion
 
-    if isfield(ASL_parms, 'Q') && isfield(ASL_parms.Q, 'EchoTime')
-        x.Q.EchoTime = ASL_parms.Q.EchoTime;
-    end
-else
-    %% PM #1543: WE MIGHT WANT TO TURN THIS AROUND
-    %% SUCH THAT WE USE json.EchoTime etc throughout here, 
-    %% which is created from x.Q if the json didn't exist
-    x.Q.EchoTime_PWI4D = JSON.Q.EchoTime;
-    x.Q.InitialPLD_PWI4D = JSON.Q.Initial_PLD;
-    x.Q.LabelingDuration_PWI4D = JSON.Q.LabelingDuration;
+if isempty(jsonASL)
```

Comment:
This is a fix for backward compatibility only.
How about we:

1. issue a warning about the missing JSON
2. try to find a legacy x.P.Path_ASL4D_parms_mat; if we find it we use this and x.Q.*, still issuing another warning

Otherwise, we have to say here how a user can run this part without having a JSON, for backward compatibility. Or simply state that we break backward compatibility ;)

Reply:
we can't find this in x.Q or in parms_mat - cause that wouldn't be valid for despiked data. 
We rewrote this function to be general and accept any input, so here, we can't anymore use x.Q.

Reply:
Yes, it's a backward compatibility issue only. I agree that we should break the backward compatibility. Shall we at the start of xASL_module_ASL do a check for data validity (e.g., issue a warning for missing JSON that this needs fixing first, and detecting _parms.mat and saying that this is old and needs conversion)?

If we do this "backward compatibility/data validity" check at the start of xASL_module_ASL, we can remove these checks from the rest of the wrappers.

Reply:
For missing JSONs and non-missing _parms.mat for ASL4D.json at the start of the ASL module, we should indeed make a new check and make a new issue for this.

However, this check here, is not about having old data. This is just a random check that the PWI4D.json was correctly calculated and nobody removed it :D So we can make this new issue, but also keep this check here to help users deal with random errors. Though we can also remove it as not really necessary.

Reply:
Fully agree. Changed that inside: d3271b25